src/main/java/seedu/address/logic/commands/ApproveLeaveCommand.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.leave.Date;
+import seedu.address.model.leave.Leave;
+import seedu.address.model.leave.Status;
+import seedu.address.model.person.ComparablePerson;
+
+/**
+ * Approves an existing leave in the leave book.
+ */
+public class ApproveLeaveCommand extends Command {
+    public static final String COMMAND_WORD = "approve-leave";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Approves leave request identified "
+            + "by the index number used in the leave book. "
+            + "The specified leave request will be approved.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1 ";
+
+    public static final String MESSAGE_APPROVE_LEAVE_SUCCESS = "Approved Leave: %1$s";
+    public static final String MESSAGE_DUPLICATE_LEAVE_APPROVE = "Leave previously approved: %1$s";
+
+    private final Index index;
+
+    /**
+     * @param index of the leave in the leave book to approve
+     */
+    public ApproveLeaveCommand(Index index) {
+        requireNonNull(index);
+
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        ObservableList<Leave> leaveList = model.getLeavesBook().getLeaveList();
```

Comment:
hi i noticed two things:
 a. this might not work with our future filters since this works based off the entire list instead of the filtered, observed list.
b. this breaks the friend of a friend law

Reply:
Good point! Thanks for pointing them out. I will look into how to update them